content/blog/2025-03-11-Alive-Peephole-Optimizations

```diff
@@ -0,0 +1,62 @@
++++
+title = "Alive: Provably Correct Peephole Optimizations"
+[extra]
+[[extra.authors]]
+name = "Annabel Baniak"
+[[extra.authors]]
+name = "Katherine Wu"
+[[extra.authors]]
+name = "Max Fan"
+[[extra.authors]]
+name = "Stephanie Ma"
++++
+
+# Alive: Provably Correct Peephole Optimizations
+A blog post about the paper ["Provably Correct Peephole Optimizations With Alive"](https://dl.acm.org/doi/pdf/10.1145/2737924.2737965).
+
+## Background / Vocab Terms, For the Uninitiated:
+
+* [LLVM](https://llvm.org/)- An SSA-based intermediate representation used in many real world contexts, especially with C programs.
+* [DSLs](https://www.geeksforgeeks.org/domain-specific-languages-in-scala/)- Domain Specific Languages, which are small programming languages specially designed for different domains or areas of study.
+* [Peephole Optimizations](https://www.geeksforgeeks.org/peephole-optimization-in-compiler-design/)- A small, simple optimization, replacing some inefficient instruction with a more efficient one, e.g. replacing a multiply instruction with a shift left.
+* [SMT solvers](https://ocamlpro.github.io/verification_for_dummies/smt/)- Solvers which take in a mathematical formula as input, and output valid variable assignments to make the expression return true (or satisfiable), when possible. These are essentially a generalization of SAT solvers: SMT formulas include variables representing integers, strings, arrays, etc. along with constraints and operators related to these theories, whereas SAT deals solely with Boolean logic.
```

Comment:
It's a tiny bit awkward to start with a "disembodied" list of terms before you jump into the prose. If you came across this blog post and wanted to learn about the Alive work, it might be kinda weird to have to start with a glossary. Do you think you might be able to just integrate the links you have here into the main text? For example, you can probably just link to "peephole optimizations" from the first occurrence of that phrase. And I don't think you need to define LLVM.